### ISSUE TITLE
job-manager: alloc segfault

### ISSUE THREAD

**Issue by chu11** [flux-framework/flux-core#3055]
while trying to reproduce #3052 hit a segfault on catalyst

```
>cat submit_priority.sh
#!/bin/bash

jobid=`flux mini submit sleep 1000`
sleep 5
flux jobs -A
sleep 5
flux job priority ${jobid} 31
sleep 5
flux jobs -A
```

```
>src/cmd/flux start ./submit_priority.sh
             JOBID USER     NAME       STATUS NTASKS NNODES  RUNTIME RANKS
       36205232128 achu     sleep           R      1      1   5.505s 0
lt-flux-broker: src/zlistx.c:531: zlistx_reorder: Assertion `handle' failed.
flux-job: 36205232128: No such file or directory
Abort (core dumped)
```

```
(gdb) bt
#0  0x00002aaaac429387 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:55
#1  0x00002aaaac42aa78 in __GI_abort () at abort.c:90
#2  0x00002aaaac4221a6 in __assert_fail_base (fmt=0x2aaaac57dce0 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", 
    assertion=assertion@entry=0x2aaaab1b7e34 "handle", file=file@entry=0x2aaaab1c06c4 "src/zlistx.c", line=line@entry=531, 
    function=function@entry=0x2aaaab1cf066 "zlistx_reorder") at assert.c:92
#3  0x00002aaaac422252 in __GI___assert_fail (assertion=0x2aaaab1b7e34 "handle", file=0x2aaaab1c06c4 "src/zlistx.c", line=531, 
    function=0x2aaaab1cf066 "zlistx_reorder") at assert.c:101
#4  0x00002aaaab17e7d0 in zlistx_reorder () from /lib64/libczmq.so.3
#5  0x00002aaabd36ac36 in alloc_queue_reorder (alloc=<optimized out>, job=<optimized out>) at alloc.c:545
#6  0x00002aaabd36ba7c in priority_handle_request (h=0x2aaad0001270, mh=<optimized out>, msg=0x2aaad000e3c0, arg=0x2aaabd7a5a00)
    at priority.c:92
#7  0x00002aaaaacecae4 in call_handler (mh=0x2aaad000c220, msg=msg@entry=0x2aaad000e3c0) at msg_handler.c:231
#8  0x00002aaaaaced219 in dispatch_message (type=1, msg=0x2aaad000e3c0, d=0x2aaad000ade0) at msg_handler.c:267
#9  handle_cb (r=0x2aaad000a5b0, hw=<optimized out>, revents=<optimized out>, arg=0x2aaad000ade0) at msg_handler.c:367
#10 0x00002aaaaad16925 in ev_invoke_pending (loop=0x2aaad000a5d0) at ev.c:3372
#11 0x00002aaaaad1a477 in ev_run (loop=0x2aaad000a5d0, flags=0) at ev.c:3775
#12 0x00002aaaaacebc73 in flux_reactor_run (r=r@entry=0x2aaad000a5b0, flags=flags@entry=0) at reactor.c:126
#13 0x00002aaabd365e83 in mod_main (h=0x2aaad0001270, argc=<optimized out>, argv=<optimized out>) at job-manager.c:131
#14 0x000000000040b7cc in module_thread (arg=0x66b7a0) at module.c:214
#15 0x00002aaaab87dea5 in start_thread (arg=0x2aaabd7a6700) at pthread_create.c:307
#16 0x00002aaaac4f18dd in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111

```

**Comment by grondo** [comment]
This crash (and a proposed fix) were described in https://github.com/flux-framework/flux-core/issues/3051#issuecomment-660191960

**Comment by chu11** [comment]
ahh didn't realize

**Comment by grondo** [comment]
Oh, I just meant that as a comment here. Still good to have an issue open on it, sorry!

**Comment by grondo** [comment]
I think this was fixed by #3072
